ocaml/doc/jsapi.ml

```diff
@@ -83,14 +87,39 @@ let generate_files destdir =
       "{'cls': '" ^ obj.name ^ "', 'obj_changes': " ^ Jsonrpc.to_string (rpc_of_changes_t obj_changes) ^ ", 'field_changes': " ^ Jsonrpc.to_string (rpc_of_changes_t field_changes) ^ ", 'msg_changes': " ^ Jsonrpc.to_string (rpc_of_changes_t msg_changes) ^ "}"
     in
     let release_info = String.concat ", " (List.map search_obj objs) in
-    let fname = rel ^ ".json" in
+    let fname = rel.code_name ^ ".json" in
     Stdext.Unixext.write_string_to_file (Filename.concat api_dir fname) ("release_info = [" ^ release_info ^ "]")
   in
   List.iter changes_in_release release_order;
-  let release_list = String.concat ", " (List.map (fun s -> "'" ^ s ^ "'") release_order) in
+  let release_list = String.concat ", " (List.map (fun s -> "'" ^ s.code_name ^ "'") release_order) in
   Stdext.Unixext.write_string_to_file (Filename.concat api_dir "releases.json") ("releases = [" ^ release_list ^ "]")
 
+let json_releases =
+  let json_of_rel x = `O [
+    "code_name", `String x.code_name;
+    "version_major", `Float (float_of_int x.version_major);
+    "version_minor", `Float (float_of_int x.version_minor);
+    "branding", `String x.branding;
+    "xs_branding", `String x.xs_branding;
+    ]
+  in
+  `O [ "releases", `A (List.map json_of_rel release_order) ]
+
+let render_template template_file json output_file =
+  let lines_read = Stdext.Unixext.read_lines template_file in
+  let templ = Mustache.of_string (String.concat "\n" lines_read) in
+  let rendered = Mustache.render templ json in
+  let out_chan = open_out output_file in
+  finally (fun () -> Printf.fprintf out_chan "%s" rendered)
+          (fun () -> close_out out_chan)
+
+let populate_releases templates_dir dest_dir=
+  let infile x = Filename.concat templates_dir x in
+  let outfile x = Filename.concat dest_dir x in
+  let render x = render_template (infile (fst x)) json_releases (outfile (snd x)) in
```

Comment:
When `x` is a pair, it would be better to define `let render (infile, outfile) = ` and to use `infile` and `outfile`. Does it have to be a pair? These could be also passed as separate parameters.